### ISSUE TITLE
ExecutionFailed: Cannot invoke RelationshipDataAccessor.properties because relationshipCursor is null

### ISSUE THREAD

**Issue by CelineWuest** [neo4j/neo4j#13150]
When running the following query:
```cypher
MERGE ()-[:A]->({x:0}) RETURN EXISTS { (:!(B&C))-[{x:0}]->() }
```

I encountered the following error:
```
Neo4jError: Neo.DatabaseError.Statement.ExecutionFailed (Cannot invoke "org.neo4j.internal.kernel.api.RelationshipDataAccessor.properties(org.neo4j.internal.kernel.api.PropertyCursor, org.neo4j.storageengine.api.PropertySelection)" because "relationshipCursor" is null)
```

I believe the query mentioned above is semantically and syntactically correct and thus no error should be thrown here.

I encountered this issue when testing queries against the **Neo4j 5.6.0 enterprise version** in a Docker container running **alpine v.3**.

### Steps to reproduce
Run the following queries and observe it throws an error:
```cypher
MERGE ()-[:A]->({x:0}) RETURN EXISTS { (:!(B&C))-[{x:0}]->() }
```

### Expected behavior
The query should run successfully

### Actual behavior
The query fails with the error message `[..]ExecutionFailed (Cannot invoke "[..] RelationshipDataAccessor.properties([..]PropertyCursor, [..]PropertySelection)" because "relationshipCursor" is null)`.

**Comment by InverseFalcon** [comment]
Not a runtime bug, the query executes on my system no matter what runtime I specify. Something else is going on here.

**Comment by loveleif** [comment]
Thanks @DominicWuest! I also can't reproduce this issue. Are you sure there's no copy paste error in the query?

**Comment by CelineWuest** [comment]
Thanks a lot for looking into all my reports @loveleif!
Sorry, while reducing the originally generated query, the exception kept being thrown, but after restarting my neo4j docker container, it ran successfully. It seems like my method of quickly clearing the database does not wipe everything.
I have reduced the query again, to a point where it errors even on a fresh install and updated the issue accordingly!

**Comment by loveleif** [comment]
Awesome, easily reproduced now. I've created a ticket in our internal bug tracker for further investigation ([link](https://trello.com/c/yq9YDxHB)).

**Comment by loveleif** [comment]
Note, most probably the same issue as https://github.com/neo4j/neo4j/issues/13147.

**Comment by sherfert** [comment]
@DominicWuest 
We have found and fixed the issue and the fix will be part of 5.9
I could not reproduce the issue on 4.4.
